pilot/pkg/proxy/envoy/v1/watcher.go

```diff
@@ -211,22 +224,27 @@ func watchCerts(ctx context.Context, certsDirs []string, watchFileEventsFn watch
 	watchFileEventsFn(ctx, fw.Event, minDelay, updateFunc)
 }
 
-func generateCertHash(h hash.Hash, certsDir string, files []string) {
+// generateCertHash generates the hash value based on the contents of cert files in the certs directory.
+// It returns false if requireFiles is true and any of the files does not exist.
+func generateCertHash(h hash.Hash, certsDir string, files []string, requireFiles bool) bool {
 	if _, err := os.Stat(certsDir); os.IsNotExist(err) {
-		return
+		return !requireFiles
```

Comment:
BTW I don't think you need this `Stat` call: [`ReadFile` will return an error if the file doesn't exist](https://github.com/golang/go/blob/025134b0d1b5e5ea2c5216fe4ccb060ab1bea59a/src/io/ioutil/ioutil.go#L53-L56).